planner/optimize.go

```diff
@@ -89,7 +90,10 @@ func GetExecuteForUpdateReadIS(node ast.Node, sctx sessionctx.Context) infoschem
 		}
 		if preparedPointer, ok := vars.PreparedStmts[execID]; ok {
 			if preparedObj, ok := preparedPointer.(*core.CachedPrepareStmt); ok && preparedObj.ForUpdateRead {
-				return domain.GetDomain(sctx).InfoSchema()
+				if vars.LocalTemporaryTables != nil {
```

Comment:
It can cause a bug：If `vars.LocalTemporaryTables == nil` , it will return nil

Reply:
my mistake. i will modify it